### PR TITLE
enable poppler support on Windows builds

### PR DESCRIPTION
**Pull request by gillins**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/gdal-feedstock/issues/235 and also fixes https://github.com/conda-forge/gdal-feedstock/pull/228 (hopefully) by explicitly running the tests.

### CODE REVIEW

**Comment by gillins** [comment]
conda-forge-admin, please rerender

File: recipe/set_bld_opts.bat [hunk 1]
```diff
@@ -66,6 +73,11 @@ set BLD_OPTS=%WIN64% ^
     GEOTIFF_INC="-I%LIBRARY_INC%" ^
     GEOTIFF_LIB=%LIBRARY_LIB%\geotiff_i.lib ^
     LIBICONV_INCLUDE="-I%LIBRARY_INC% -DICONV_CONST= " ^
-    LIBICONV_LIBRARY="%LIBRARY_LIB%\iconv.lib"
+    LIBICONV_LIBRARY="%LIBRARY_LIB%\iconv.lib" ^
+    POPPLER_ENABLED=YES ^
+    POPPLER_CFLAGS="-I%LIBRARY_INC%\poppler -DPOPPLER_0_20_OR_LATER= -DPOPPLER_0_23_OR_LATER= -DPOPPLER_0_58_OR_LATER= -DPOPPLER_HAS_OPTCONTENT= -DPOPPLER_BASE_STREAM_HAS_TWO_ARGS= " ^
+    POPPLER_MAJOR_VERSION=%POPPLER_MAJOR_VERSION% ^
+    POPPLER_MINOR_VERSION=%POPPLER_MINOR_VERSION% ^
+    POPPLER_LIBS="%LIBRARY_LIB%\poppler.lib %LIBRARY_LIB%\freetype.lib advapi32.lib"
```
Comment (gillins) [inline recipe/set_bld_opts.bat]: I have to link to the static lib on Windows as not all the needed symbols are `export`ed by `poppler`. This means I have to list any other needed libs that aren't already linked to `gdal`...

Comment (gillins) [reply recipe/set_bld_opts.bat]: Update: managed to enable shared libs in https://github.com/conda-forge/poppler-feedstock/pull/37. Seems to link ok now.

**Comment by gillins** [comment]
@ocefpaf any idea about the OSX error (below)? I think this is ready to go apart from that.

```
+ gdal_grid --version
dyld: Library not loaded: @rpath/libgeos-3.4.2.dylib
  Referenced from: /Users/distiller/miniconda3/conda-bld/gdal-split_1548368535284/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/libspatialite.7.dylib
  Reason: image not found
/Users/distiller/miniconda3/conda-bld/gdal-split_1548368535284/test_tmp/run_test.sh: line 7: 41941 Abort trap: 6           gdal_grid --version
Tests failed for libgdal-2.4.0-h89caebc_1002.tar.bz2 - moving package to /Users/distiller/miniconda3/conda-bld/broken
```

**Comment by kmuehlbauer** [comment]
@gillins GEOS 3.4.2 seems fishy, no idea why that is pulled in.

**Comment by kmuehlbauer** [comment]
@gillins @ocefpaf Should we need a rerender of `libspatialite-feedstock`? There has been something going on with the GEOS pin in the recent past.

**Comment by ocefpaf** [comment]
> @gillins @ocefpaf Should we need a rerender of `libspatialite-feedstock`? There has been something going on with the GEOS pin in the recent past.

Yep. Working on it.

**Pull request merged**